server/channel_state_test.go

```diff
@@ -0,0 +1,262 @@
+			ID:      "test",
+			StartAt: 100,
+			Users:   []string{"userA"},
+			States: []UserState{
+				{RaisedHand: 1100},
+			},
+		}
+
+		require.Equal(t, &ccs, cs.getClientState("botID"))
+	})
+}
+
+func TestCallStateGetHostID(t *testing.T) {
+	t.Run("empty", func(t *testing.T) {
+		var cs callState
+		require.Empty(t, cs.getHostID("botID"))
+	})
+
+	t.Run("singl user", func(t *testing.T) {
+		cs := &callState{
+			ID:      "test",
+			StartAt: 100,
+			Users: map[string]*userState{
+				"userA": {
+					JoinAt:     1000,
+					RaisedHand: 1100,
+				},
+			},
+		}
+
+		require.Equal(t, "userA", cs.getHostID("botID"))
+	})
+
+	t.Run("multiple users", func(t *testing.T) {
+		cs := &callState{
+			ID:      "test",
+			StartAt: 100,
+			Users: map[string]*userState{
+				"userA": {
+					JoinAt:     1000,
+					RaisedHand: 1100,
+				},
+				"userB": {
+					JoinAt:  800,
+					Unmuted: true,
+				},
+				"userC": {
+					JoinAt:  1100,
+					Unmuted: true,
+				},
+			},
+		}
+
+		require.Equal(t, "userB", cs.getHostID("botID"))
+	})
+
+	t.Run("skip botID", func(t *testing.T) {
+		cs := &callState{
+			ID:      "test",
+			StartAt: 100,
+			Users: map[string]*userState{
+				"botID": {
+					JoinAt: 800,
+				},
+				"userA": {
+					JoinAt:     1000,
+					RaisedHand: 1100,
+				},
+				"userB": {
+					JoinAt:  1100,
+					Unmuted: true,
+				},
+				"userC": {
+					JoinAt:  1200,
+					Unmuted: true,
+				},
+			},
+		}
+
+		require.Equal(t, "userA", cs.getHostID("botID"))
+	})
+}
+
+func TestChannelStateClone(t *testing.T) {
+	t.Run("nil", func(t *testing.T) {
+		var cs *channelState
+		require.Nil(t, cs.Clone())
+	})
+
+	t.Run("empty", func(t *testing.T) {
+		var cs channelState
+		cloned := cs.Clone()
+		require.Empty(t, cloned)
+		require.NotSame(t, &cs, cloned)
+	})
+
+	t.Run("nil call", func(t *testing.T) {
+		cs := channelState{
+			NodeID:  "nodeID",
+			Enabled: true,
+		}
+		cloned := cs.Clone()
+		require.Equal(t, cs, *cloned)
+		require.NotSame(t, &cs, cloned)
+	})
+
+	t.Run("with empty call", func(t *testing.T) {
+		cs := channelState{
+			NodeID:  "nodeID",
+			Enabled: true,
+			Call:    &callState{},
+		}
+		cloned := cs.Clone()
+		require.Equal(t, cs, *cloned)
+		require.NotSame(t, &cs, cloned)
+		require.NotSame(t, cs.Call, cloned.Call)
+	})
+
+	t.Run("with non-empty call", func(t *testing.T) {
+		cs := &channelState{
+			NodeID:  "nodeID",
+			Enabled: true,
+			Call: &callState{
+				Users: map[string]*userState{
+					"userA": {
+						JoinAt:  1000,
+						Unmuted: true,
+					},
+					"userB": {
+						JoinAt:  1000,
+						Unmuted: true,
+					},
+					"userC": {
+						JoinAt:  1000,
+						Unmuted: true,
+					},
+				},
+				Sessions: map[string]struct{}{
+					"userA": {},
+					"userC": {},
+					"userB": {},
+				},
+			},
+		}
+
+		cloned := cs.Clone()
+		require.Equal(t, cs, cloned)
+		require.NotSame(t, cs, cloned)
+
+		require.Condition(t, func() bool {
+			return cs.Call != cloned.Call
+		})
```

Comment:
Because those fields are pointers and I want to make sure they are not pointing to the original object which is a problem we had before with `channelState.Call` (it was a bug really but luckily didn't affect anything).